testbed/integration/prometheus/e2e_test.go

```diff
@@ -0,0 +1,144 @@
+//go:build e2e
+
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	"go.opentelemetry.io/collector/receiver/receivertest"
+
+	"github.com/Dynatrace/dynatrace-otel-collector/internal/k8stest"
+)
+
+const (
+	testKubeConfig = "/tmp/kube-config-collector-e2e-testing"
+)
+
+// TestE2E_PrometheusNodeExporter tests the "Scrape data from Prometheus" use case
+// See: https://docs.dynatrace.com/docs/shortlink/otel-collector-cases-prometheus
+func TestE2E_PrometheusNodeExporter(t *testing.T) {
+	testDir := filepath.Join("testdata")
+
+	k8sClient, err := k8stest.NewK8sClient(testKubeConfig)
+	require.NoError(t, err)
+
+	// Create the namespace specific for the test
+	nsFile := filepath.Join(testDir, "namespace.yaml")
+	buf, err := os.ReadFile(nsFile)
+	require.NoErrorf(t, err, "failed to read namespace object file %s", nsFile)
+	nsObj, err := k8stest.CreateObject(k8sClient, buf)
+	require.NoErrorf(t, err, "failed to create k8s namespace from file %s", nsFile)
+
+	testNs := nsObj.GetName()
+	defer func() {
+		require.NoErrorf(t, k8stest.DeleteObject(k8sClient, nsObj), "failed to delete namespace %s", testNs)
+	}()
+
+	// Install Prometheus Node exporter
+	installPrometheusNodeExporter()
+
+	metricsConsumer := new(consumertest.MetricsSink)
+	shutdownSinks := startUpSinks(t, metricsConsumer)
+	defer shutdownSinks()
+
+	testID := uuid.NewString()[:8]
+	collectorObjs := k8stest.CreateCollectorObjects(t, k8sClient, testID, filepath.Join(testDir, "collector"))
+	defer func() {
+		for _, obj := range collectorObjs {
+			require.NoErrorf(t, k8stest.DeleteObject(k8sClient, obj), "failed to delete object %s", obj.GetName())
+		}
+	}()
+
+	wantEntries := 2 // Minimal number of metric requests to wait for.
+	waitForData(t, wantEntries, metricsConsumer)
+
+	expectedMetrics := []string{
+		"otelcol_process_memory_rss", "scrape_duration_seconds", "scrape_samples_post_metric_relabeling",
+	}
+
+	scanForServiceMetrics(t, metricsConsumer, "opentelemetry-collector", expectedMetrics)
+}
+
+func installPrometheusNodeExporter() error {
+	testDir := filepath.Join("testdata", "prometheus")
+	nsFile := filepath.Join(testDir, "install.sh")
+	cmd, err := exec.Command("/bin/bash", nsFile).Output()
+
+	if err != nil {
+		return fmt.Errorf("Failed to install Prometheus node exporter Helm chart %s", err)
+	}
+
+	fmt.Print(string(cmd))
+
```

Comment:
```suggestion
```
We should probably remove this, right?

Reply:
It is helpful because it prints the results from the help commands, which show when the Prometheus node exporter is ready. So I think keeping it makes sense.

Reply:
That's fair. This is minor, but could you add a comment above this line repeating what you just described? Raw print statements stick out, so that way we know why this is here in the future.